modules/core/arrow-extras-data/src/main/kotlin/arrow/data/fingertree/internal/Node.kt

```diff
@@ -0,0 +1,22 @@
+package arrow.data.fingertree.internal
+
+internal sealed class Node<T> {
+
+  abstract fun toList(): List<T>
+
+  data class Branch2<T>(val first: T, val second: T) : Node<T>() {
```

Comment:
First and second never get accessed directly outside of the class. Maybe it's faster to just store the list directly on the `fromList` constructor and desambiguate on size.